api/src/main/java/io/minio/MakeBucketArgs.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * MinIO Java SDK for Amazon S3 Compatible Cloud Storage, (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.minio;
+
+/** Bucket Arguments to be used to create Bukcet. */
+public class MakeBucketArgs extends BucketArgs {
+  private final boolean objectLock;
+
+  private MakeBucketArgs(Builder builder) {
+    super(builder);
+    this.objectLock = builder.objectLock;
+  }
+
+  /** Return true if object lock functionality is enabled */
+  public boolean objectLock() {
+    return objectLock;
+  }
+
+  /** Return new Builder */
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static final class Builder extends BucketArgs.Builder<Builder> {
+    private boolean objectLock;
+
+    public Builder() {}
+
+    public Builder(MakeBucketArgs args) {
```

Comment:
You could remove `Builder(XXXArgs args)` constructor everywhere and will be added on demand.